pkg/image/airgap.go

```diff
@@ -831,14 +836,20 @@ func pushOCIArtifact(opts imagetypes.PushOCIArtifactOptions) error {
 			Password: opts.Registry.Password,
 		}),
 	}
-	repository.PlainHTTP = true
 
-	_, err = oras.Copy(context.TODO(), orasFS, opts.Tag, repository, opts.Tag, oras.DefaultCopyOptions)
-	if err != nil {
-		return errors.Wrap(err, "failed to copy")
+	_, err1 := oras.Copy(context.TODO(), orasFS, opts.Tag, repository, opts.Tag, oras.DefaultCopyOptions)
+	if err1 == nil {
+		return nil
 	}
 
-	return nil
+	// try again with plain http
```

Comment:
i thought we're going to check the existence of the embedded cluster registry tls secret here too? wouldn't this pollute the registry logs with red-herrings?

Reply:
ah I believe I misunderstood the ask here. i can try to find a good way to get a k8s client into this function or at the very least a new parameter specifying plainHTTP true/false.

on the logs piece, i'm not seeing additional logs in docker distribution, but that may depend on the registry implementation

Reply:
ultimately deciding to leave as-is as TLS is the go-forward in embedded-cluster so it is very unlikely that the non-TLS path will be reached. if it does and that still fails, both errors will be returned for debugging.